src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -90,14 +91,26 @@ export default class EthereumBlockchainService implements BlockchainService {
   }
 
   /**
-   * Takes current gas price and attempt number and returns new gas price with 10% increase per attempt
-   * @param currentGas
+   * Takes current gas price and attempt number and returns new gas price with 10% increase per attempt.
+   * If this isn't the first attempt, also ensures that the new gas is at least 10% greater than the
+   * previous attempt's gas, even if the gas price on chain has gone down since then. This is because
+   * retries of the same transaction (using the same nonce) need to have a gas price at least 10% higher
+   * than any previous attempts or the transaction will fail.
+   * @param currentGasEstimate
    * @param attempt
+   * @param previousGas
    */
-  static increaseGasPricePerAttempt(currentGas: BigNumber, attempt: number): BigNumber {
-    const tenPercent = currentGas.div(10)
+  static increaseGasPricePerAttempt(currentGasEstimate: BigNumber, attempt: number, previousGas: BigNumberish | undefined): BigNumber {
+    const tenPercent = currentGasEstimate.div(10)
     const additionalGas = tenPercent.mul(attempt)
-    return currentGas.add(additionalGas)
+    const newGas = currentGasEstimate.add(additionalGas)
+    if (attempt == 0 || !previousGas) {
```

Comment:
Can `previousGas` here be 0 or 1? Is this better as `previousGas != undefined`?

Reply:
I imagine gas can never be 0, and it's hard to imagine it ever being 1, but doesn't hurt to be precise.  Updated.